internal/incident/incidents.go

```diff
@@ -25,38 +27,122 @@ var (
 )
 
 // LoadOpenIncidents loads all active (not yet closed) incidents from the database and restores all their states.
-// Returns error ony database failure.
+// Returns error on any database failure.
 func LoadOpenIncidents(ctx context.Context, db *icingadb.DB, logger *logging.Logger, runtimeConfig *config.RuntimeConfig) error {
 	logger.Info("Loading all active incidents from database")
 
-	var objectIDs []types.Binary
-	err := db.SelectContext(ctx, &objectIDs, `SELECT object_id FROM incident WHERE "recovered_at" IS NULL`)
-	if err != nil {
-		logger.Errorw("Failed to load active incidents from database", zap.Error(err))
+	g, ctx := errgroup.WithContext(ctx)
 
-		return errors.New("failed to fetch open incidents")
-	}
+	incidents := make(chan *Incident)
+	g.Go(func() error {
+		defer close(incidents)
 
-	for _, objectID := range objectIDs {
-		obj, err := object.LoadFromDB(ctx, db, objectID)
+		rows, err := db.QueryxContext(ctx, db.BuildSelectStmt(new(Incident), new(Incident))+` WHERE "recovered_at" IS NULL`)
 		if err != nil {
-			logger.Errorw("Failed to retrieve incident object from database", zap.Error(err))
-			continue
+			return err
 		}
+		// In case the incidents in the loop below are successfully traversed, rows is automatically closed and an
+		// error is returned (if any), making this rows#Close() call a no-op. Escaping from this function unexpectedly
+		// means we have a more serious problem, so in either case just disregard the error here.
+		defer func() { _ = rows.Close() }()
+
+		for rows.Next() {
+			i := NewIncident(db, nil, runtimeConfig, nil)
+			if err := rows.StructScan(i); err != nil {
+				return err
+			}
 
-		incident, _, err := GetCurrent(ctx, db, obj, logger, runtimeConfig, false)
-		if err != nil {
-			continue
+			select {
+			case incidents <- i:
+			case <-ctx.Done():
+				return ctx.Err()
+			}
 		}
 
-		incident.RetriggerEscalations(&event.Event{
-			Time:    time.Now(),
-			Type:    event.TypeInternal,
-			Message: "Incident reevaluation at daemon startup",
-		})
-	}
+		return rows.Err()
+	})
+
+	g.Go(func() error {
+		bulks := com.Bulk(ctx, incidents, db.Options.MaxPlaceholdersPerStatement, com.NeverSplit[*Incident])
+
+		for {
+			select {
+			case <-ctx.Done():
+				return ctx.Err()
+			case bulk, ok := <-bulks:
+				if !ok {
+					return nil
+				}
+
+				g.Go(func() error {
+					chunkLen := len(bulk)
+					objectIds := make([]types.Binary, chunkLen)
+					incidentIds := make([]int64, chunkLen)
+					incidentsById := make(map[int64]*Incident, chunkLen)
+					incidentsByObjId := make(map[string]*Incident, chunkLen)
+
+					for k, i := range bulk {
+						incidentsById[i.Id] = i
+						incidentsByObjId[i.ObjectID.String()] = i
+
+						objectIds[k] = i.ObjectID
+						incidentIds[k] = i.Id
+					}
```

Comment:
Those loop variable are somewhat unexpected/confusing. I'd just go for `for i, incident := range bulk`. (Or you could initialize the slices as `make([]T, 0, chunkLen)` and keep using `append()`.)